File: library/src/main/java/de/timfreiheit/mozart/utils/BitmapHelper.java
package de.timfreiheit.mozart.utils;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;

import java.io.InputStream;

public class BitmapHelper {

    public static final int MAX_ART_WIDTH_PX = 800;
    public static final int MAX_ART_HEIGHT_PX = 480;

    public static final int MAX_ART_WIDTH_ICON_PX = 128;
    public static final int MAX_ART_HEIGHT_ICON_PX = 128;

    // Max read limit that we allow our input stream to mark/reset.
    private static final int MAX_READ_LIMIT_PER_IMG = 1024 * 1024;

    public static Bitmap scaleBitmap(Bitmap src, int maxWidth, int maxHeight) {
        double scaleFactor = Math.min(
                ((double) maxWidth) / src.getWidth(), ((double) maxHeight) / src.getHeight());
        return Bitmap.createScaledBitmap(src,
                (int) (src.getWidth() * scaleFactor), (int) (src.getHeight() * scaleFactor), false);
    }

    public static Bitmap scaleBitmap(int scaleFactor, InputStream is) {
        // Get the dimensions of the bitmap
        BitmapFactory.Options bmOptions = new BitmapFactory.Options();

        // Decode the image file into a Bitmap sized to fill the View
        bmOptions.inJustDecodeBounds = false;
        bmOptions.inSampleSize = scaleFactor;

        return BitmapFactory.decodeStream(is, null, bmOptions);
    }

    public static int findScaleFactor(int targetW, int targetH, InputStream is) {
        // Get the dimensions of the bitmap
        BitmapFactory.Options bmOptions = new BitmapFactory.Options();
        bmOptions.inJustDecodeBounds = true;
        BitmapFactory.decodeStream(is, null, bmOptions);
        int actualW = bmOptions.outWidth;
        int actualH = bmOptions.outHeight;

        // Determine how much to scale down the image
        return Math.min(actualW / targetW, actualH / targetH);
    }

}

File: mozart/src/main/java/de/timfreiheit/mozart/utils/BitmapHelper.kt
package de.timfreiheit.mozart.utils

import android.graphics.Bitmap
import android.graphics.BitmapFactory

import java.io.InputStream


internal fun scaleBitmap(src: Bitmap, maxWidth: Int, maxHeight: Int): Bitmap {
    val scaleFactor = Math.min(
            maxWidth.toDouble() / src.width, maxHeight.toDouble() / src.height)
    return Bitmap.createScaledBitmap(src,
            (src.width * scaleFactor).toInt(), (src.height * scaleFactor).toInt(), false)
}

internal fun scaleBitmap(scaleFactor: Int, `is`: InputStream): Bitmap {
    // Get the dimensions of the bitmap
    val bmOptions = BitmapFactory.Options()

    // Decode the image file into a Bitmap sized to fill the View
    bmOptions.inJustDecodeBounds = false
    bmOptions.inSampleSize = scaleFactor

    return BitmapFactory.decodeStream(`is`, null, bmOptions)
}

internal fun findScaleFactor(targetW: Int, targetH: Int, `is`: InputStream): Int {
    // Get the dimensions of the bitmap
    val bmOptions = BitmapFactory.Options()
    bmOptions.inJustDecodeBounds = true
    BitmapFactory.decodeStream(`is`, null, bmOptions)
    val actualW = bmOptions.outWidth
    val actualH = bmOptions.outHeight

    // Determine how much to scale down the image
    return Math.min(actualW / targetW, actualH / targetH)
}
